### ISSUE TITLE
[Search] problem with lowercase option values

### ISSUE THREAD

**Issue by remyma** [Sylius/Sylius#2958]
I started a dev project with sylius. I wrote my custom data fixtures based on sylius ones and load products with color options with values like : "red", "black", ...

I noticed that facetted search based on ORM did not work with my option values. The problem comes from that I use lowercase values and OrmFinder expects a first capitalized letter.

=> if I use options like "Red", "Black" it is ok.

Fix could be to replace OrmFinder l.271
```php
// filtering on an array of values
if (is_array($tags[strtolower($key)]) && in_array(ucfirst($separateFilter[$key]), $tags[strtolower($key)])) {
    $result[] = $facet['itemId'];
    // got the value, I don't want to move into more checks
    continue;
 }
```

with something like this : 
```php
// filtering on an array of values
if (is_array($tags[strtolower($key)]) && in_array(strtolower($separateFilter[$key]), array_map('strtolower', $tags[strtolower($key)]))) 
{
    $result[] = $facet['itemId'];
    // got the value, I don't want to move into more checks
    continue;
 }
```
But I'm not sure it's the best way to handle that.

Any ideas ?

**Comment by nmarmon** [comment]
Hi,

I try to create some products and and taxonomies, for example, and then I click on the button of search in the products section.  For example, I create a product named "Mouse", and in the field for the search I put "mouse". Finally I click on the magnifying glass button. And sylius found the product.

Where do you have the problem exactly?

**Comment by remyma** [comment]
It is not about products nor taxonomies. The problem occurs with facet name when you try to filter a search by option

**Comment by kereso** [comment]
Hi!

I can confirm this issue too: in order for the filtering to work, I have to define my products' attributes with their first letter capitalized. I want to filter products of a taxon. The product list is loaded in:

/Sylius/Bundle/CoreBundle/Controller/ProductController.php, method indexByTaxonAction()

then it is filtered out in

/Sylius/Bundle/SearchBundle/Finder/OrmFinder.php

as already described by remyma.

If I define product attributes with their first letter capitalized, everything is just fine. However, my site is in hungarian and I cannot capitalize the first letter of words, only those of sentences.

![taste-filtering](https://cloud.githubusercontent.com/assets/6926708/10387569/266e8746-6e6b-11e5-92af-9d3bfe332e9d.png)

In my opinion a good solution would be to convert the values to be compared to lowercase/uppercase before compare them. However, I do not know the rationale behind the original implementation so there might be a more appropriate solution.

**Comment by pjedrzejewski** [comment]
Fixed in #3410. Thanks for reporting!
